mathics/eval/sympy.py

```diff
@@ -0,0 +1,76 @@
+"""
+Evaluation of sympy functions
+"""
+
+import sys
+from queue import Queue
+from threading import Thread
+
+import sympy
+
+import mathics.eval.tracing as tracing
+from mathics.core.convert.sympy import from_sympy, to_numeric_sympy_args
+
+
+def eval_sympy_unconstrained(self, z, evaluation):
+    """
+    Evaluate an expression converting it to Sympy
+    and back to Mathics.
+    """
+    sympy_args = to_numeric_sympy_args(z, evaluation)
+    if self.sympy_name is None:
+        return
+    sympy_fn = getattr(sympy, self.sympy_name)
+    try:
+        return from_sympy(tracing.run_sympy(sympy_fn, *sympy_args))
+    except Exception:
+        return
+
+
+def eval_sympy_with_timeout(self, z, evaluation):
+    """
+    Evaluate an expression converting it to Sympy,
+    and back to Mathics.
+    This version put the evaluation in a thread,
+    and check each some time if the evaluation
+    reached a timeout.
+    """
+
+    if evaluation.timeout is None:
+        return eval_sympy_unconstrained(self, z, evaluation)
+
+    def _thread_target(queue) -> None:
+        try:
+            result = eval_sympy_unconstrained(self, z, evaluation)
+            queue.put((True, result))
+        except BaseException:
+            exc_info = sys.exc_info()
+            queue.put((False, exc_info))
+
+    queue = Queue(maxsize=1)  # stores the result or exception
+
+    def evaluate():
```

Comment:
@mmatera where/how is `evaluate()` used?

Reply:
@mmatera For what it's worth, I removed the `evaluate()` function and the tests still run fine. Can we remove this code?

Reply:
Yes. I forgot to remove it. I moved its code to `eval_sympy_unconstrained()`, so we can use it when `Threads` are not available.